components/usage/pkg/db/workspace_instance.go

```diff
@@ -87,3 +89,20 @@ func ListWorkspaceInstancesInRange(ctx context.Context, conn *gorm.DB, from, to
 
 	return instances, nil
 }
+
+func NewAttributionID(entity, identifier string) AttributionID {
```

Comment:
Nit: I guess Go can infer the type here, but it looks a bit inconsistent without the explicit type:

```suggestion
func NewAttributionID(entity string, identifier string) AttributionID {
```

Reply:
This is not inference, this is standard style of give. When there are multiple arguments of the same type, you only need to specify the type on the last one.

Reply:
Thanks! Today I learned. (I guess a few lines below, the type can also be factored out.)

Reply:
Interesting. This works (compiles) but I've not really noticed this pattern in other people's code before. Perhaps it's because it's less common to return two values of the same type from a function